mkFit/MkBuilderEndcap.cc

```diff
@@ -130,13 +130,13 @@ void MkBuilderEndcap::begin_event(Event* ev, EventTmp* ev_tmp, const char* build
     }
   }
 
-  if (Config::readCmsswSeeds==false) m_event->seedTracks_.resize(simtracks.size());
+  if (Config::readCmsswSeeds==false) m_event->seedTracks_ = m_event->simTracks_;
 }
 
-inline void MkBuilderEndcap::fit_one_seed_set_endcap(TrackVec& simtracks, int itrack, int end, MkFitter *mkfp)
+inline void MkBuilderEndcap::fit_one_seed_set_endcap(TrackVec& seedtracks, int itrack, int end, MkFitter *mkfp)
 {
-  mkfp->SetNhits(2); //note this is 2 instead of 3 since we ignore PXB1
-  mkfp->InputTracksAndHits(simtracks, m_event->layerHits_, itrack, end);
+  mkfp->SetNhits(Config::nlayers_per_seed); //note this is 2 instead of 3 since we ignore PXB1
```

Comment:
In case people missed this point, I made Config::nlayers_per_seed an extern, with a default = 3 for the barrel: https://github.com/kmcdermo/mictest/blob/seeds-tbb/Config.cc#L28

If the flag "--endcap-test" is used in ./mkFit, then Config::nlayers_per_seed gets set to 2: https://github.com/kmcdermo/mictest/blob/seeds-tbb/mkFit/mkFit.cc#L529-L532